src/pages/Stats/index.tsx

```diff
@@ -0,0 +1,279 @@
+    stats_api.searchParams.append('apiKey', ethplorerApiKey)
+
+    const statsReq = await fetch(stats_api.href)
+    const statsRes = await statsReq.json()
+
+    if (!statsRes.hasOwnProperty('error')) {
+      const price = statsRes.price
+      const userGrumpyValueInUsd = balance * price.rate
+
+      setPrice('$' + price.rate.toFixed(11))
+      setMarketCap(
+        '$' +
+          price.marketCapUsd.toLocaleString(undefined, {
+            maximumFractionDigits: 0,
+          })
+      )
+      setGrumpyUsdValue('$' + formatPriceUsd(userGrumpyValueInUsd))
+    }
+  }
+
+  async function getWallet() {
+    if (account) {
+      const balance = await getGrumpyBalance(account)
+      const tx = await getGrumpyTransaction(account, balance)
+      getGrumpyStats(balance)
+
+      setGrumpyBalance(balance)
+
+      setTotalIn(tx.totalIn)
+      setTotalOut(tx.totalOut)
+      setRedistributedAmount(tx.redistribution)
+      setGrumpyBalanceWithoutRedistribution(tx.balanceWithoutRedistribution)
+    }
+  }
+
+  async function getGrumpyBalance(account: string) {
+    const balance_api = new URL('https://api.etherscan.io/api')
+
+    balance_api.searchParams.append('module', 'account')
+    balance_api.searchParams.append('action', 'tokenbalance')
+    balance_api.searchParams.append('contractaddress', grumpyContractAddress)
+    balance_api.searchParams.append('address', account)
+    balance_api.searchParams.append('tag', 'latest')
+    balance_api.searchParams.append('apikey', ethescanApiKey)
+
+    const balanceReq = await fetch(balance_api.href)
+    const balanceRes = await balanceReq.json()
+    const balance = parseFloat(balanceRes.result)
+
+    return balance
+  }
+
+  async function getGrumpyTransaction(account: string, balance: number) {
+    const transactions_api = new URL('https://api.etherscan.io/api')
+
+    transactions_api.searchParams.append('module', 'account')
+    transactions_api.searchParams.append('action', 'tokentx')
+    transactions_api.searchParams.append('contractaddress', grumpyContractAddress)
+    transactions_api.searchParams.append('address', account)
+    transactions_api.searchParams.append('page', '1')
+    transactions_api.searchParams.append('offset', '10000')
+    transactions_api.searchParams.append('apikey', ethescanApiKey)
+
+    const transactionReq = await fetch(transactions_api.href)
+    const transactionRes = await transactionReq.json()
+    const transaction = transactionRes.result
+
+    let totalIn = 0.0
+    let totalOut = 0.0
+
+    for (const item of transaction) {
+      if (item.to === account.toLowerCase()) {
+        totalIn += parseFloat(item.value)
+      } else {
+        totalOut += parseFloat(item.value)
+      }
+    }
+
+    const balanceWithoutRedistribution = totalIn - totalOut
+    const redistribution = balance - balanceWithoutRedistribution
+
+    return { totalIn, totalOut, redistribution, balanceWithoutRedistribution }
+  }
+
+  useEffect(() => {
+    getWallet()
```

Comment:
I get a warning:
```
src/pages/Stats/index.tsx
  Line 164:6:  React Hook useEffect has a missing dependency: 'getWallet'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```

But this isn't major since it runs anyway.